src/views/Wiser.vue

```diff
@@ -52,7 +91,19 @@ export default {
     }
   },
   mounted () {
-    // this.popupActive = !this.skipintro
+    this.portraitMode = window.matchMedia('(orientation: portrait)').matches && window.innerWidth < 768
+  },
+  beforeCreate: function () {
+    let self = this
+    $backend.getNextCampaign()
+      .then((res) => {
+        self.campaignId = res.id
+        let selectContent = CampaignTexts[res.name]
+        self.text = selectContent.start
+        console.debug(selectContent.id, 'text loaded')
+        // Trigger loading images from the selected campaign
+        self.$refs.imageVote.nextImagePair()
```

Comment:
Why was the ImageVote.nextImagePair() moved here instead of the ImageVote's lifecycle mounted fn?

Reply:
Because we needed a delay for loading the content, at least in my experience the update did not seem to propagate reactively. Do you have a better suggestion?

Reply:
I would argue that campaign is anyway being passed in as a prop. The ImageVote component should depend on that for re-rendering itself.

Currently the ImageVote expects a campaign prop, but doesn't use it at all. `campaign_id` is used is in the `nextImagePair` fn of backend.js, but there we read it from localstorage. We can change the `nextImagePair` fn to accept an optional campaign_id param.

The present code works fine, but in its present form ImageVote component is a leaky abstraction, with its parent component knowing the internal workings of it.

Reply:
Agreed, corrected using the reactive prop & `watch`/`mounted` methods.